File: LeetCode/Easy/Permutation Difference between Two Strings/3146.js
/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var findPermutationDifference = function(s, t) {
    let sum = 0;

    const idxMap = {};
    for (let i = 0; i < s.length; i++) {
        idxMap[s.charAt(i)] = i;
    }

    for (let j = 0; j < t.length; j++) {
        sum += Math.abs(idxMap[t.charAt(j)] - j);
    }

    return sum;
};

File: LeetCode/Easy/Permutation Difference between Two Strings/3146.java
class Solution {
    public int findPermutationDifference(String s, String t) {
        int sum = 0;

        int[] idxArr = new int[26];
        for (int i = 0; i < s.length(); i++) {
            idxArr[s.charAt(i) - 'a'] = i;
        }

        for (int j = 0; j < t.length(); j++) {
            sum += Math.abs(j - idxArr[t.charAt(j) - 'a']);
        }

        return sum;
    }
}
